### PR TITLE
Updating wwwAuthenticateParsing to handle v2.0

### PR DESCRIPTION
**Pull request by trwalke**
Fixes #
https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/4416

**Changes proposed in this request**
Updating wwwAuthenticateParsing to handle v2.0 path

**Testing**
unit testing

**Performance impact**

**Documentation**
- [ ] All relevant documentation is updated.

### CODE REVIEW

**Comment by trwalke** [comment]
@bgavrilMS you mentioned in the issue the uri "https://[login.microsoftonline.com//common/v2.0/.well-known/openid-configuration](https://login.microsoftonline.com//common/v2.0/.well-known/openid-configuration)"

Is this supposed to be tested as well? It doesn't seem like an authorization uri.

File: src/client/Microsoft.Identity.Client/WwwAuthenticateParameters.cs [hunk 1]
```diff
@@ -420,14 +420,14 @@ internal static WwwAuthenticateParameters CreateWwwAuthenticateParameters(IDicti
 
             if (values.TryGetValue("authorization_uri", out string value))
             {
-                wwwAuthenticateParameters.Authority = value.Replace("/oauth2/authorize", string.Empty);
+                wwwAuthenticateParameters.Authority = value.Replace("/v2.0", string.Empty).Replace("/oauth2/authorize", string.Empty);
```
Comment (trwalke) [inline src/client/Microsoft.Identity.Client/WwwAuthenticateParameters.cs:423]: can we assume that "/oauth2" will always come before "/authorize"

**Review by bgavrilMS — APPROVED** [review]

**Comment by bgavrilMS** [comment]
> @bgavrilMS you mentioned in the issue the uri "https://[login.microsoftonline.com//common/v2.0/.well-known/openid-configuration](https://login.microsoftonline.com//common/v2.0/.well-known/openid-configuration)"
> 
> Is this supposed to be tested as well? It doesn't seem like an authorization uri.

https://[login.microsoftonline.com//common/v2.0/.well-known/openid-configuration](https://login.microsoftonline.com//common/v2.0/.well-known/openid-configuration) just shows you how the authorization endpoint is advertised by AAD.

File: src/client/Microsoft.Identity.Client/WwwAuthenticateParameters.cs (hunk 1 above)
Comment (gladjohn) [inline src/client/Microsoft.Identity.Client/WwwAuthenticateParameters.cs:423]:
```suggestion
                // Strip "/v2.0" and "/oauth2/authorize" to normalize the authority URL.
                // Ensures compatibility and consistency across API versions.
                wwwAuthenticateParameters.Authority = value.Replace("/v2.0", string.Empty).Replace("/oauth2/authorize", string.Empty);
```

**Review by gladjohn — APPROVED** [review]

**Pull request closed without merging**